include/range/v3/view_interface.hpp

```diff
@@ -297,6 +297,32 @@ namespace ranges
             {
                 return Slice{}(detail::move(derived()), offs.from, offs.to);
             }
+            /// Returns a reference to the element at specified location pos, with bounds checking.
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(Same<D, Derived>() && RandomAccessRange<D>() && SizedRange<D>())>
+            auto at(range_difference_type_t<D> n) ->
+                decltype(std::declval<D &>().begin()[n])
+            {
+                using size_type = range_size_type_t<Derived>;
+                if (n < 0 || size_type(n) >= ranges::size(derived()))
+                {
+                    throw std::out_of_range("view_interface::at");
+                }
+                return derived().begin()[n];
```

Comment:
That is strange. Are you sure the warning is produced on the line with the explicit conversion? Isn't it from somewhere else? I suppose the range-v3 library may have a few places where it implicitly changes signedness.

[This](https://gist.github.com/FloopCZ/f9ff686883056096b28a09f25be52dc4) seems to work as expected.